src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs

```diff
@@ -210,48 +210,36 @@ internal void InferBoundPropertyModelPrefixes(ControllerModel controllerModel)
                 }
             }
         }
-        
+
         // internal for testing
         internal void InferParameterModelPrefixes(ActionModel actionModel)
         {
             foreach (var parameter in actionModel.Parameters)
             {
-                if (parameter.BindingInfo != null &&
-                    parameter.BindingInfo.BinderModelName == null &&
-                    parameter.BindingInfo.BindingSource != null &&
-                    !parameter.BindingInfo.BindingSource.IsGreedy)
+                var bindingInfo = parameter.BindingInfo;
+                if (bindingInfo?.BindingSource != null &&
+                    bindingInfo.BinderModelName == null &&
+                    !bindingInfo.BindingSource.IsGreedy &&
+                    IsComplexTypeParameter(parameter))
```

Comment:
@rynowak I kept the `IsComplexTypeParameter(...)` part separate. Please LMK your thoughts.

Reply:
Seems fine.